src/prefect/engine/results/secret_result.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any
+
+import prefect
+from prefect.engine.result import Result
+
+
+class SecretResult(Result):
+    """
+    Hook for storing and retrieving sensitive task results from a Secret store. Only
+    intended to be used for Secret Tasks - each call to "read" will actually rerun
+    the underlying secret task and re-retrieve the secret value.
+
+    Args:
+        - value (Any): the underlying value this Result should represent
+    """
+
+    def __init__(
+        self, secret_task: "prefect.tasks.secrets.Secret", **kwargs: Any
+    ) -> None:
+        self.secret_task = secret_task
+        kwargs.setdefault("filepath", secret_task.name)
+        super().__init__(**kwargs)
+
+    def read(self, filepath: str) -> Result:
```

Comment:
Update it to location while you're here :)

Reply:
Nooooooo